src/js_native_api_v8.cc

```diff
@@ -773,6 +815,177 @@ napi_status NAPI_CDECL napi_get_last_error_info(
   return napi_ok;
 }
 
+#define HANDLE_ERRORS_VECTOR(errors, vec)                                      \
+  {                                                                            \
+    if (errors == nullptr) {                                                   \
+      for (const std::string& error : vec)                                     \
+        fprintf(stderr, "%s\n", error.c_str());                                \
+    } else {                                                                   \
+      *errors = node::Malloc<char*>(vec.size() + 1);                           \
+      if (errors == nullptr) return napi_generic_failure;                      \
+      char** cur_error = *errors;                                              \
+      for (const std::string& error : vec) {                                   \
+        *(cur_error++) = strdup(error.c_str());                                \
+      }                                                                        \
+      *cur_error = nullptr;                                                    \
+    }                                                                          \
+  }
+
+napi_status NAPI_CDECL napi_create_platform(int argc,
+                                            char** argv,
+                                            int exec_argc,
+                                            char** exec_argv,
+                                            char*** errors,
```

Comment:
With the ABI-safe APIs we cannot allocate memory in one module and free it in another. E.g. in our Microsoft Office apps we use custom memory allocations with additional safety checks. The memory allocated in embedded Node.JS cannot be deallocated in Office code that may embed it. To be ABI-safe all memory must be allocated and freed in the same module.
For example, in the Node-API we often use the Finalizer callback that supports such single memory ownership pattern. In COM we use AddRef/Release to enforce this pattern too. 
In this case we must at least return a pointer to the finalizer function that must be called by `napi_create_platform` user after they finish processing errors, or have a special `napi_free_string_array`.
My personal preference is to have a special `write_error` callback that we call to write the errors to the hosting app. In that case we would not need to allocate memory and just give them `data()` and `length()` from our stored `std::string`. It will be up to the hosting app how to handle it - it can either print or store them. The `write_error` callback could be called multiple times - once per each error string. There is no need to "terminate" the list - the end of the `napi_create_platform` is a natural `termination`. The `write_error` callback must accept the `void* hostingData` which must be passed to the `napi_create_platform` to let the hosting application to encapsulate its context for the `write_error` callback.